test/test_export_vm.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+from unittest.mock import MagicMock, call
+
+import pytest
+from dateutil.tz import tzutc
+
+from exasol_script_languages_developer_sandbox.lib.export_vm.run_export_vm import export_vm
+from exasol_script_languages_developer_sandbox.lib.export_vm.vm_disk_image_format import all_vm_disk_image_formats, \
+    VmDiskImageFormat
+from exasol_script_languages_developer_sandbox.lib.vm_bucket.vm_slc_bucket import STACK_NAME
+from test.conftest import DEFAULT_ASSET_ID
+
+INSTANCE_ID = "test-instance"
+TEST_ROLE_ID = 'VM-SLC-Bucket-VMImportRole-TEST'
+TEST_BUCKET_ID = 'vm-slc-bucket-vmslcbucket-TEST'
+
+
+def get_only_vm_stack_side_effect(stack_name: str):
+    if stack_name == STACK_NAME:
+        # The following is a snapshot from calling AwsAccess(a).get_all_stack_resources("VM-SLC-Bucket") on a running
+        # cloudformation stack
+        return [{'LogicalResourceId': 'VMImportRole',
+                 'PhysicalResourceId': TEST_ROLE_ID,
+                 'ResourceType': 'AWS::IAM::Role',
+                 'LastUpdatedTimestamp': datetime.datetime(2022, 8, 11, 17, 15, 20, 380000, tzinfo=tzutc()),
+                 'ResourceStatus': 'CREATE_COMPLETE', 'DriftInformation': {'StackResourceDriftStatus': 'NOT_CHECKED'}},
+                {'LogicalResourceId': 'VMSLCBucket', 'PhysicalResourceId': TEST_BUCKET_ID,
+                 'ResourceType': 'AWS::S3::Bucket',
+                 'LastUpdatedTimestamp': datetime.datetime(2022, 8, 11, 17, 14, 55, 63000, tzinfo=tzutc()),
+                 'ResourceStatus': 'CREATE_COMPLETE', 'DriftInformation': {'StackResourceDriftStatus': 'NOT_CHECKED'}}]
+    else:
+        raise ValueError(f"Unexpected parameter:{stack_name}")
+
+
+TEST_AMI_ID = "AMI-IMAGE-12345"
+
+
+def call_counter(func):
```

Comment:
woh, this is a bit crazy, maybe a small explanation would be good

Reply:
I added some explanation